mllib/src/main/scala/org/apache/spark/mllib/optimization/GradientDescent.scala

```diff
@@ -144,6 +145,21 @@ class GradientDescent private[spark] (private var gradient: Gradient, private va
  */
 @DeveloperApi
 object GradientDescent extends Logging {
+  /** Java-friendly version of [[runMiniBatchSGD()]] */
+  def runMiniBatchSGD(
```

Comment:
I don't think we need to make this Java-friendly.  It's a DeveloperApi, and I've never heard of Java users wanting to build algorithms off of our GradientDescent.  Could you remove this please?